lib/utils.js

```diff
@@ -53,3 +53,8 @@ export async function loadGetInitialProps (Component, ctx) {
   }
   return props
 }
+
+export function getLocationOrigin () {
+  const { protocol, hostname, port } = window.location
+  return `${protocol}//${hostname}${port ? ':' + port : ''}`
```

Comment:
I don't agree, strong concatenation simply shouldn't be used. But anyway it's a detail.